util.go

```diff
@@ -2,24 +2,46 @@ package cuckoo
 
 import (
 	"encoding/binary"
-	"math/rand"
+	"unsafe"
 
-	metro "github.com/dgryski/go-metro"
+	"github.com/zeebo/wyhash"
+	"github.com/zeebo/xxh3"
+)
+
+var (
+	// plus 1 to prevent out of range.
+	altHash = [maxFingerprint + 1]uint{}
+	rng     wyhash.SRNG
 )
 
 // randi returns either i1 or i2 randomly.
 func randi(i1, i2 uint) uint {
-	if rand.Int31()%2 == 0 {
+	// it's faster than mod, but the result is the almost same.
+	if uint32(uint64(uint32(rng.Uint64()))*uint64(2)>>32) == 0 {
 		return i1
 	}
 	return i2
 }
+func getEndian() (endian binary.ByteOrder) {
+	var i int = 0x1
+	bs := (*[int(unsafe.Sizeof(0))]byte)(unsafe.Pointer(&i))
+	if bs[0] == 0 {
+		return binary.BigEndian
+	} else {
+		return binary.LittleEndian
+	}
+}
+func init() {
+	b := make([]byte, 2)
+	endian := getEndian()
```

Comment:
What's the advantage in deriving endianes? Hard-coding either big or little for all platforms should do the same afaik.

Reply:
Undo, I don't see the benefit.